authority/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+// Disable the following two lints since they originate from an external macro (namely decl_storage)
+#![allow(clippy::string_lit_as_bytes)]
+
+use frame_support::{
+	decl_error, decl_module,
+	dispatch::PostDispatchInfo,
+	ensure,
+	traits::{EnsureOrigin, Get},
+	weights::{FunctionOf, GetDispatchInfo, Pays},
+	Parameter,
+};
+use frame_system::{self as system};
+use orml_traits::{DelayedDispatchTime, DispatchId, Scheduler};
+use sp_runtime::{
+	traits::{BadOrigin, CheckedAdd, CheckedSub, Dispatchable},
+	RuntimeDebug,
+};
+use sp_std::prelude::*;
+
+mod mock;
+mod tests;
+
+#[derive(PartialEq, Eq, Clone, RuntimeDebug)]
+pub struct DelayedOrigin<BlockNumber, Origin> {
+	pub delay: BlockNumber,
+	pub origin: Origin,
+}
+
+/// Origin for the authority module.
+pub type Origin<T> = DelayedOrigin<<T as system::Trait>::BlockNumber, <T as frame_system::Trait>::Origin>;
+
+pub struct EnsureDelayed<Delay, Inner, BlockNumber>(sp_std::marker::PhantomData<(Delay, Inner, BlockNumber)>);
+
+impl<
+		O: Into<Result<DelayedOrigin<BlockNumber, O>, O>> + From<DelayedOrigin<BlockNumber, O>>,
+		Delay: Get<BlockNumber>,
+		Inner: EnsureOrigin<O>,
+		BlockNumber: PartialOrd,
+	> EnsureOrigin<O> for EnsureDelayed<Delay, Inner, BlockNumber>
+{
+	type Success = ();
```

Comment:
Something like this to keep Success type (not sure syntax) 

`type Success = Inner::Success;`